src/java.base/share/classes/jdk/internal/foreign/NativeMemorySegmentImpl.java

```diff
@@ -147,6 +150,14 @@ public void cleanup() {
         return segment;
     }
 
+    private static long allocateMemoryWrapper(long size) {
```

Comment:
This is required as we no longer call Bits::reserveMemory, so it is possible when passing `Long.MAX_VALUE` as a size parameter, to get a IAE from Unsafe. Note that I think this was probably a latent issue - while the behavior changed for `Long.MAX_VALUE` with this patch, I believe using `Long.MAX_VALUE -1` would have created similar issues with the original code. Point is: `Unsafe` might fail because the size is too big - and we should handle that.